server/routes/return-of-service.js

```diff
@@ -2,17 +2,50 @@ const express = require('express');
 const keycloak = require('../keycloak');
 const logger = require('../logger.js');
 const { asyncMiddleware, applyMiddleware } = require('../error-handler.js');
-const { rosError } = require('../constants');
+const { rosError, rosFieldUpdate } = require('../constants');
 const { CreateReturnOfServiceSchema, validate } = require('../validators');
 const { getParticipantByID } = require('../services/participants');
 const {
   makeReturnOfServiceStatus,
   getReturnOfServiceStatuses,
+  updateReturnOfServiceSite,
+  updateReturnOfServiceDate,
+  updateReturnOfServiceStartDate,
 } = require('../services/return-of-service');
 const { getSiteDetailsById, getDetailsBySiteId } = require('../services/employers');
 
 const router = express.Router();
 
+const validateParticipant = async (participantId, actionType) => {
+  const [participant] = await getParticipantByID({ id: participantId });
+  if (!participant) {
+    logger.error({
+      action: actionType,
+      message: `Participant ${participantId} not found`,
+    });
+    return null;
+  }
+  return participant;
+};
+
+const getRosErrorMessage = (message) => {
```

Comment:
If we're getting a message, should we pass in something like `type` rather than `message`?